src/vs/editor/common/config/editorOptions.ts

```diff
@@ -4156,6 +4165,11 @@ class EditorSuggest extends BaseEditorOption<EditorOption.suggest, ISuggestOptio
 					default: true,
 					markdownDescription: nls.localize('editor.suggest.showDeprecated', "When enabled IntelliSense shows `deprecated`-suggestions.")
 				},
+				'editor.suggest.allowMidWordMatch': {
+					type: 'boolean',
+					default: false,
+					markdownDescription: nls.localize('editor.suggest.allowMidWordMatch', "When enabled suggestions allow a first characer to be matched on a non-capital letter. This will provide more search results that may not be as accurate.")
+				},
```

Comment:
It's not just about capital character but also about character after separators, like `_` or `.`. I have started a team discussion about how to talk about this and will come back with a decision